frontend/src/__tests__/cypress/cypress/e2e/pipelines/Pipelines.cy.ts

```diff
@@ -116,9 +116,155 @@ describe('Pipelines', () => {
+          deleted
+            ? buildMockPipelineVersions([])
+            : buildMockPipelineVersions([initialMockPipelineVersion]),
+        );
+      },
+    );
+
+    // Wait for the pipelines table to load
+    pipelinesTable.find();
+
+    // Check pipeline version
+    pipelinesTable.toggleExpandRowByName(initialMockPipeline.name);
+    pipelinesTable.toggleCheckboxByRowName(initialMockPipelineVersion.name);
+
+    // Delete the selected version
+    pipelinesGlobal.findDeleteButton().click();
+    pipelineDeleteModal.shouldBeOpen();
+    pipelineDeleteModal.findInput().type(initialMockPipelineVersion.name);
+    pipelineDeleteModal.findSubmitButton().click();
+
+    cy.wait('@deleteVersion');
+
+    // Check empty version state
+    pipelinesTable.shouldHaveNoVersions(initialMockPipeline.name);
+  });
+
+  it('delete pipeline and versions', () => {
+    let deleted = false;
+    const mockPipeline1 = buildMockPipeline({ name: 'Test pipeline 1' });
+    const mockPipeline1Version1 = buildMockDefaultPipelineVersion(mockPipeline1);
+
+    const mockPipeline2 = buildMockPipeline({ name: 'Test pipeline 2' });
+    const mockPipeline2Version1 = buildMockDefaultPipelineVersion(mockPipeline2);
+    const mockPipeline2Version2 = buildMockPipelineVersion({
+      id: 'test-pipeline-2-new-version',
+      name: 'Test pipeline 2 new version',
+      resource_references: [
+        {
+          key: { type: ResourceTypeKF.PIPELINE, id: mockPipeline2.id },
+          relationship: RelationshipKF.OWNER,
+        },
+      ],
+    });
+
+    cy.intercept(
+      {
+        pathname: '/api/proxy/apis/v1beta1/pipelines',
+      },
+      (req) => {
+        if (deleted) {
+          req.reply(buildMockPipelines([mockPipeline2]));
+        } else {
+          req.reply(buildMockPipelines([mockPipeline1, mockPipeline2]));
+        }
+      },
+    );
+    cy.intercept(
+      {
+        method: 'POST',
+        pathname: '/api/proxy/apis/v1beta1/pipeline_versions',
+      },
+      (req) => {
+        const response = deleted
+          ? { [mockPipeline2.id]: buildMockPipelineVersions([mockPipeline2Version2]) }
+          : {
+              [mockPipeline1.id]: buildMockPipelineVersions([mockPipeline1Version1]),
+              [mockPipeline2.id]: buildMockPipelineVersions([
+                mockPipeline2Version1,
+                mockPipeline2Version2,
+              ]),
+            };
+        req.reply(response[req.body.queryParams['resource_key.id']]);
+      },
+    );
+
+    cy.intercept(
+      {
+        pathname: `/api/proxy/apis/v1beta1/pipelines/${mockPipeline1.id}`,
+        method: 'POST',
+      },
+      { body: {} },
+    ).as('deletePipeline');
+    cy.intercept(
+      {
+        pathname: `/api/proxy/apis/v1beta1/pipeline_versions/${mockPipeline2Version1.id}`,
+        method: 'POST',
+      },
+      (req) => {
+        deleted = true;
+        req.reply({ body: {} });
+      },
+    ).as('deleteVersion');
+    initIntercepts();
```

Comment:
nit: I would intercept these prior to the pipeline intercepts above, just to be consistent with the other tests and the order in which they happen.